config/configtelemetry/configtelemetry.go

```diff
@@ -40,6 +40,9 @@ const (
 
 const UseOpenTelemetryForInternalMetrics = false
 
+// indicates if the collector version tag should be added to all telemetry metrics
+const AddCollectorVersionTag = true
```

Comment:
You might move this into `service/telemetry.go` directly since it'll be the only place using it and there is another PR up that will move the OTel flag there as well.